server/plugin/virtual_agent.go

```diff
@@ -256,12 +267,84 @@ func (p *Plugin) ProcessResponse(data []byte) error {
 			if _, err = p.DMWithAttachments(userID, p.CreateOutputCardAttachment(&data)); err != nil {
 				return err
 			}
+		case *OutputImage:
+			post, err := p.CreateOutputImagePost(res, userID)
+			if err != nil {
+				return err
+			}
+
+			if _, err = p.dm(userID, post); err != nil {
+				return err
+			}
 		}
 	}
 
 	return nil
 }
 
+func (p *Plugin) CreateOutputImagePost(body *OutputImage, userID string) (*model.Post, error) {
+	channel, appErr := p.API.GetDirectChannel(userID, p.botUserID)
+	if appErr != nil {
+		p.API.LogInfo("Couldn't get bot's DM channel", "user_id", userID, "error", appErr.Message)
+		return nil, appErr
+	}
+
+	resp, err := http.Get(body.Value)
+	if err != nil {
+		p.API.LogInfo("Error in getting file data from link", "error", err.Error())
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	data, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		p.API.LogInfo("Error in reading file data", "error", err.Error())
+		return nil, err
+	}
+
+	linkContents := strings.Split(body.Value, "/")
+
+	if len(linkContents) < 1 {
+		p.API.LogInfo(InvalidImageLinkError)
+		return nil, errors.New(InvalidImageLinkError)
+	}
+
+	completeFilename := linkContents[len(linkContents)-1]
+
+	filenameContents := strings.Split(completeFilename, ".")
+
+	ContentTypeInHeaders := ""
```

Comment:
```suggestion
	contentTypeInHeaders := ""
```

Reply:
Fixed [here](https://github.com/Brightscout/mattermost-plugin-servicenow-virtual-agent-private/pull/6/files#diff-0f5f6f38127a9ea36ffecf95673cc3105903094fb27fa91e0774291963b3ddf9)

Reply:
You shared a link of the private repo, this change should also be in the public repo.

Reply:
Yeah its in the public repo as well